colord-1.4.0 does not build

I'm still struggling to get a buildable GNOME 3.25.90. My current hangup:

```
*** Configuring colord *** [1/199]
meson --prefix /home/mcatanzaro/releases/gnome-apps-3.25.90/install --libdir lib -Denable-bash-completion=false -Denable-vala=true --buildtype=debugoptimized -Ddisable_gtkdoc=true /home/mcatanzaro/releases/gnome-apps-3.25.90/colord-1.4.0
The Meson build system
Version: 0.41.2
Source dir: /home/mcatanzaro/releases/gnome-apps-3.25.90/colord-1.4.0
Build dir: /home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0
Build type: native build
Project name: colord
Native c compiler: ccache cc (gcc 7.1.1)
Appending LDFLAGS from environment: '-L/home/mcatanzaro/releases/gnome-apps-3.25.90/install/lib '
Build machine cpu family: x86_64
Build machine cpu: x86_64
Compiler for c supports argument -fstack-protector-strong: YES
Compiler for c supports argument -Waggregate-return: YES
Compiler for c supports argument -Wunused: YES
Compiler for c supports argument -Warray-bounds: YES
Compiler for c supports argument -Wcast-align: YES
Compiler for c supports argument -Wclobbered: YES
Compiler for c supports argument -Wdeclaration-after-statement: YES
Compiler for c supports argument -Wempty-body: YES
Compiler for c supports argument -Wextra: YES
Compiler for c supports argument -Wformat=2: YES
Compiler for c supports argument -Wformat-nonliteral: YES
Compiler for c supports argument -Wformat-security: YES
Compiler for c supports argument -Wformat-signedness: YES
Compiler for c supports argument -Wignored-qualifiers: YES
Compiler for c supports argument -Wimplicit-function-declaration: YES
Compiler for c supports argument -Winit-self: YES
Compiler for c supports argument -Wmissing-declarations: YES
Compiler for c supports argument -Wmissing-format-attribute: YES
Compiler for c supports argument -Wmissing-include-dirs: YES
Compiler for c supports argument -Wmissing-noreturn: YES
Compiler for c supports argument -Wmissing-parameter-type: YES
Compiler for c supports argument -Wmissing-prototypes: YES
Compiler for c supports argument -Wnested-externs: YES
Compiler for c supports argument -Wno-discarded-qualifiers: YES
Compiler for c supports argument -Wno-missing-field-initializers: YES
Compiler for c supports argument -Wno-strict-aliasing: YES
Compiler for c supports argument -Wno-suggest-attribute=format: YES
Compiler for c supports argument -Wno-unused-parameter: YES
Compiler for c supports argument -Wold-style-definition: YES
Compiler for c supports argument -Woverride-init: YES
Compiler for c supports argument -Wpointer-arith: YES
Compiler for c supports argument -Wreturn-type: YES
Compiler for c supports argument -Wshadow: YES
Compiler for c supports argument -Wsign-compare: YES
Compiler for c supports argument -Wstrict-aliasing: YES
Compiler for c supports argument -Wstrict-prototypes: YES
Compiler for c supports argument -Wswitch-default: YES
Compiler for c supports argument -Wtype-limits: YES
Compiler for c supports argument -Wundef: YES
Compiler for c supports argument -Wuninitialized: YES
Compiler for c supports argument -Wunused-but-set-variable: YES
Compiler for c supports argument -Wwrite-strings: YES
Compiler for c supports argument -Wl,-z,relro: YES
Compiler for c supports argument -Wl,-z,now: YES
Found pkg-config: /usr/bin/pkg-config (1.3.8)
Native dependency gio-2.0 found: YES 2.53.5
Native dependency glib-2.0 found: YES 2.53.5
Native dependency gmodule-2.0 found: YES 2.53.5
Native dependency gio-unix-2.0 found: YES 2.53.5
Native dependency lcms2 found: YES 2.8
Native dependency sqlite3 found: YES 3.19.3
Native dependency systemd found: YES 233
Native dependency libsystemd found: YES 233
Native dependency gusb found: YES 0.2.11
Native dependency gudev-1.0 found: YES 230
Library m found: YES
Native dependency udev found: YES 233
Native dependency libudev found: YES 233
Has header "pwd.h": YES
Has header "syslog.h": YES
Has header "unistd.h": YES
Checking for function "getuid": YES
Program spotread found: YES (/usr/bin/spotread)
Program vapigen found: YES (/home/mcatanzaro/releases/gnome-apps-3.25.90/install/bin/vapigen)
Message: RUNNING THIS AS root IS NOT A GOOD IDEA SEE --with-daemon-user
Native dependency polkit-gobject-1 found: YES 0.113
Native dependency valgrind found: YES 3.13.0
Configuring config.h using configuration
Configuring cd-version.h using configuration
Program g-ir-scanner found: YES (/home/mcatanzaro/releases/gnome-apps-3.25.90/install/bin/g-ir-scanner)
Program g-ir-compiler found: YES (/home/mcatanzaro/releases/gnome-apps-3.25.90/install/bin/g-ir-compiler)
Native dependency gobject-introspection-1.0 found: YES 1.53.5
Program vapigen found: YES (/home/mcatanzaro/releases/gnome-apps-3.25.90/install/bin/vapigen)
Configuring ch-version.h using configuration
Configuring colord-session.service using configuration
Configuring org.freedesktop.ColorHelper.service using configuration
Configuring colord.service using configuration
Configuring org.freedesktop.ColorManager.service using configuration
Configuring org.freedesktop.ColorManager.conf using configuration
Program docbook2man found: YES (/usr/bin/docbook2man)
Configuring org.freedesktop.color.policy.in using configuration
Configuring 69-cd-sensors.rules using configuration
Native dependency glib-2.0 found: YES 2.53.5
Program meson_post_install.sh found: YES (/bin/sh /home/mcatanzaro/releases/gnome-apps-3.25.90/colord-1.4.0/meson_post_install.sh)
Build targets in project: 112
*** Building colord *** [1/199]
ninja
[55/219] Generating symbol file 'lib/colord/colordprivate@sha/lib/colord/libcolordprivate.so.2.0.5.symbol[65/219] Generating symbol file 'lib/colord/colordprivate@sha/lib/colord/libcolordprivate.so.2.0.5.symbol[123/219] Generating Colord-1.0.gir with a custom command.
g-ir-scanner: link: cc -o /home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/tmp-introspectwjo683tj/Colord-1.0 /home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/tmp-introspectwjo683tj/Colord-1.0.o -L. -Wl,-rpath,. -Wl,--no-as-needed -lcolord -lcolordprivate -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lgudev-1.0 -llcms2 -lm -ludev -L/home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colord -Wl,-rpath,/home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colord -L/home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colord -Wl,-rpath,/home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colord -L/home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colord -Wl,-rpath,/home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colord -L/home/mcatanzaro/releases/gnome-apps-3.25.90/install/lib -Wl,-rpath,/home/mcatanzaro/releases/gnome-apps-3.25.90/install/lib -L/home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colord -Wl,-rpath,/home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colord -L/home/mcatanzaro/releases/gnome-apps-3.25.90/install/lib -lgio-2.0 -lgobject-2.0 -Wl,--export-dynamic -lgmodule-2.0 -pthread -lglib-2.0 -L/home/mcatanzaro/releases/gnome-apps-3.25.90/install/lib
[139/219] Generating colord.vapi with a custom command.
FAILED: lib/colord/colord.vapi 
/home/mcatanzaro/releases/gnome-apps-3.25.90/install/bin/vapigen --quiet --library=colord --directory=/home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colord --metadatadir=/home/mcatanzaro/releases/gnome-apps-3.25.90/colord-1.4.0/lib/colord /home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colord/Colord-1.0.gir
Gio-2.0.gir:2736.7-2736.37: warning: Virtual method `G.AppInfo.get_commandline' conflicts with method of the same name
Gio-2.0.gir:2777.7-2777.35: warning: Virtual method `G.AppInfo.get_executable' conflicts with method of the same name
Gio-2.0.gir:29356.7-29356.31: warning: Virtual method `G.File.get_basename' conflicts with method of the same name
Gio-2.0.gir:29441.7-29441.23: warning: Virtual method `G.File.get_path' conflicts with method of the same name
Gio-2.0.gir:29451.7-29451.41: warning: Virtual method `G.File.get_relative_path' conflicts with method of the same name
Gio-2.0.gir:62514.7-62514.47: warning: Virtual method `G.Resolver.lookup_service_async' conflicts with method of the same name
Gio-2.0.gir:64944.7-64944.31: warning: Signal `G.Settings.change_event' conflicts with method of the same name
Gio-2.0.gir:85926.7-85930.24: error: `UnixSocketAddress' already contains a definition for `abstract'
Gio-2.0.gir:85769.7-85771.21: note: previous definition of `abstract' was here
[147/219] Generating Colorhug-1.0.gir with a custom command.
Package colord was not found in the pkg-config search path.
Perhaps you should add the directory containing `colord.pc'
to the PKG_CONFIG_PATH environment variable
Package 'colord', required by 'virtual:world', not found
g-ir-scanner: link: cc -o /home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/tmp-introspectv_m42x_k/Colorhug-1.0 /home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/tmp-introspectv_m42x_k/Colorhug-1.0.o -L. -Wl,-rpath,. -Wl,--no-as-needed -lcolorhug -lcolorhug -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lgusb -lusb-1.0 -llcms2 -lm -ludev -L/home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colorhug -Wl,-rpath,/home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colorhug -L/home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colorhug -Wl,-rpath,/home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colorhug -L/home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colord -Wl,-rpath,/home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colord -L/home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colorhug -Wl,-rpath,/home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colorhug -L/home/mcatanzaro/releases/gnome-apps-3.25.90/install/lib -Wl,-rpath,/home/mcatanzaro/releases/gnome-apps-3.25.90/install/lib -L/home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colorhug -Wl,-rpath,/home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colorhug -L/home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colord -Wl,-rpath,/home/mcatanzaro/.cache/jhbuild/build/colord-1.4.0/lib/colord -L/home/mcatanzaro/releases/gnome-apps-3.25.90/install/lib -lgio-2.0 -lgobject-2.0 -Wl,--export-dynamic -lgmodule-2.0 -pthread -lglib-2.0 -L/home/mcatanzaro/releases/gnome-apps-3.25.90/install/lib
<unknown>:: Warning: Colorhug: (ErrorQuarkFunction)device_error_quark: ch_device_error_quark: Couldn't find corresponding enumeration
releases/gnome-apps-3.25.90/colord-1.4.0/lib/colorhug/ch-device-queue.c:961: Warning: Colorhug: ch_device_queue_get_calibration: argument calibration: Unresolved type: 'CdMat3x3*'
releases/gnome-apps-3.25.90/colord-1.4.0/lib/colorhug/ch-device-queue.c:1011: Warning: Colorhug: ch_device_queue_set_calibration: argument calibration: Unresolved type: 'CdMat3x3*'
releases/gnome-apps-3.25.90/colord-1.4.0/lib/colorhug/ch-device-queue.c:1072: Warning: Colorhug: ch_device_queue_set_calibration_ccmx: argument ccmx: Unresolved type: 'CdIt8*'
[148/219] Generating ProPhotoRGB.icc with a custom command.
ninja: build stopped: subcommand failed.
*** Error during phase build of colord: ########## Error running ninja *** [1/199]
```

I tried including commit 1c9c972a2a1fc16716e4be84d150d5a9ceb29fe2 and I think that got rid of the unresolved type warnings at the bottom, but the build still failed. I think the main problem is this one:

```
Gio-2.0.gir:85926.7-85930.24: error: `UnixSocketAddress' already contains a definition for `abstract'
Gio-2.0.gir:85769.7-85771.21: note: previous definition of `abstract' was here
```

--- a/lib/colord/meson.build
+++ b/lib/colord/meson.build
@@ -205,6 +205,7 @@ libcolord_typelib = libcolord_girtarget[1]
 if get_option('enable-vala')
 libfoo_vapi = gnome.generate_vapi('colord',
   sources: libcolord_girtarget[0],
+  packages: ['gio-2.0'],
   install: true,
 )
 endif

